libs/geometry/include/geometry/TangentSpaceMesh.h

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_GEOMETRY_TANGENTSPACEMESH_H
+#define TNT_GEOMETRY_TANGENTSPACEMESH_H
+
+#include <math/quat.h>
+#include <math/vec3.h>
+#include <math/vec4.h>
+#include <utils/compiler.h>
+
+namespace filament {
+namespace geometry {
+
+struct TangentSpaceMeshInput;
+struct TangentSpaceMeshOutput;
+
+/**
+ * For building Filament-style TANGENTS buffers given an input mesh.
+ * Supercedes the implementation in SurfaceOrientation.h
+ */
+class UTILS_PUBLIC TangentSpaceMesh {
```

Comment:
Since this is a public header file, maybe denote that this is still a WIP.

Reply:
Actually I would not make it public until it's ready.

Reply:
Done